lib/hiera/backend/eyaml/encryptors/gpg.rb

```diff
@@ -94,14 +103,20 @@ def self.find_recipients
 
           def self.encrypt plaintext
             gnupghome = self.option :gnupghome
-            GPGME::Engine.home_dir = gnupghome
             debug("GNUPGHOME is #{gnupghome}")
 
-            ctx = GPGME::Ctx.new
-
             recipients = self.find_recipients
             debug("Recipents are #{recipients}")
 
+            unless defined?(GPGME)
+              RubyGpg.config.homedir = gnupghome if gnupghome
+              return RubyGpg.encrypt_string(plaintext, recipients)
+            end
+
+            GPGME::Engine.home_dir = gnupghome
+
+            ctx = GPGME::Ctx.new
+
             raise RecoverableError, 'No recipients provided, don\'t know who to encrypt to' if recipients.empty?
```

Comment:
This should go before the RubyGpg use so that this error is thrown when appropriate.